lib/rules/indent.js

```diff
@@ -807,6 +892,96 @@ module.exports = {
             });
         }
 
+        /**
+        * Set the current unknown node if not set
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function setCurrentUnknownNode(node) {
+            if (KNOWN_NODES.has(node.type)) {
+                return;
+            }
+
+            if (currentUnknownNode === null) {
+                currentUnknownNode = node;
+            }
+        }
+
+        /**
+        * On exit of unknown node append it to list of outermost unknown nodes
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function saveOutermostUnknownNode(node) {
+            if (KNOWN_NODES.has(node.type)) {
+                return;
+            }
+
+            if (currentUnknownNode === node) {
+                unknownNodes.push(node);
+                currentUnknownNode = null;
+            }
+        }
+
+        /**
+        * Ignore first token of each line of within an unknown node if its
+        * offset does not depend on another token's offset. If the first token
+        * of a line's offset depends on the offset of another token, make the
+        * last dependent offset depend on the first token of its line.
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function ignoreUnknownNode(node) {
+            const startLine = node.loc.start.line;
+            const endLine = node.loc.end.line;
+
+            for (let x = startLine; x <= endLine; x++) {
+                const firstTokenOfLine = tokenInfo.firstTokensByLineNumber.get(x);
+
+                if (!firstTokenOfLine) {
+                    continue;
+                }
+
+                const tokenInNodeRange = (node.range[0] <= firstTokenOfLine.range[0]) &&
+                    (node.range[1] >= firstTokenOfLine.range[1]);
+
+                if (!tokenInNodeRange) {
+                    continue;
+                }
+
+
+                const dependency = offsets.getFirstDependency(firstTokenOfLine);
+
+                if (!dependency) {
```

Comment:
Nitpick: If this `if` statement has an `else` branch anyway, it would be clearer to use `if (dependency)` and swap the branches rather than `if (!dependency)`.

I'll make a separate PR to enable [`no-negated-condition`](http://eslint.org/docs/rules/no-negated-condition).